homeassistant/components/motioneye/config_flow.py

```diff
@@ -0,0 +1,127 @@
+        errors = {}
+        if user_input is None:
+            entry = self.context.get(CONF_CONFIG_ENTRY)
+            user_input = entry.data if entry else {}
+        else:
+            try:
+                # Cannot use cv.url validation in the schema itself, so
+                # apply extra validation here.
+                cv.url(user_input[CONF_URL])
+            except vol.Invalid:
+                errors["base"] = "invalid_url"
+            else:
+                client = create_motioneye_client(
+                    user_input[CONF_URL],
+                    admin_username=user_input.get(CONF_ADMIN_USERNAME),
+                    admin_password=user_input.get(CONF_ADMIN_PASSWORD),
+                    surveillance_username=user_input.get(CONF_SURVEILLANCE_USERNAME),
+                    surveillance_password=user_input.get(CONF_SURVEILLANCE_PASSWORD),
+                )
+
+                try:
+                    await client.async_client_login()
+                except MotionEyeClientConnectionError:
+                    errors["base"] = "cannot_connect"
+                except MotionEyeClientInvalidAuthError:
+                    errors["base"] = "invalid_auth"
+                except MotionEyeClientRequestError:
+                    errors["base"] = "unknown"
+                else:
+                    entry = self.context.get(CONF_CONFIG_ENTRY)
+                    if (
+                        self.context.get(CONF_SOURCE) == SOURCE_REAUTH
+                        and entry is not None
+                    ):
+                        self.hass.config_entries.async_update_entry(
+                            entry, data=user_input
+                        )
+                        # Need to manually reload, as the listener won't have been
+                        # installed because the initial load did not succeed (the reauth
+                        # flow will not be initiated if the load succeeds).
+                        await self.hass.config_entries.async_reload(entry.entry_id)
+                        out = self.async_abort(reason="reauth_successful")
+                        return out
+
+                    out = self.async_create_entry(
+                        title=f"{user_input[CONF_URL]}",
+                        data=user_input,
+                    )
+                    return out
+
+        out = self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_URL, default=user_input.get(CONF_URL, "")): str,
+                    vol.Optional(
+                        CONF_ADMIN_USERNAME, default=user_input.get(CONF_ADMIN_USERNAME)
+                    ): str,
+                    vol.Optional(
+                        CONF_ADMIN_PASSWORD, default=user_input.get(CONF_ADMIN_PASSWORD)
+                    ): str,
+                    vol.Optional(
+                        CONF_SURVEILLANCE_USERNAME,
```

Comment:
Why do we need both admin and surveillance accounts? Can't we just use admin for everything? Or get surveillance credentials using the admin account ?

Reply:
I'm afraid not. The admin account is required for many API calls ([example](https://github.com/ccrisan/motioneye/blob/dev/motioneye/handlers.py#L314)). Whereas only the surveillance account can be used to access the actual stream.

There is no way to retrieve credentials via the API.